nixos/modules/installer/tools/nixos-rebuild.sh

```diff
@@ -281,7 +282,7 @@ fi
 
 # Resolve the flake.
 if [[ -n $flake ]]; then
-    flake=$(nix flake info --json "${extraBuildFlags[@]}" "${lockFlags[@]}" -- "$flake" | jq -r .url)
+    flake=$($NIX flake info --json "${extraBuildFlags[@]}" "${lockFlags[@]}" -- "$flake" | jq -r .url)
```

Comment:
This also requires the `flakes` experimental feature. In any case, I think the user should affirmatively enable this in `nix.conf` themselves. Otherwise it kind of defeats the point of `experimental-features`, which is to make people aware of the fact that they're using an experimental feature.